addons/auth_oauth/models/auth_oauth.py

```diff
@@ -21,3 +23,12 @@ class AuthOAuthProvider(models.Model):
     css_class = fields.Char(string='CSS class', default='fa fa-fw fa-sign-in text-primary')
     body = fields.Char(required=True, help='Link text in Login Dialog', translate=True)
     sequence = fields.Integer(default=10)
+
+    @api.model
+    def _neutralize(self):
+        super()._neutralize()
+        providers = self.search([('enabled', '=', True)])
+        providers.write({'enabled': False})
+        for provider in providers:
+            _logger.info('Oauth provider `%s` neutralized', provider.name)
+        assert self.search([('enabled', '=', True)]) == self.env['auth.oauth.provider'], "Not all oauth providers were neutralized !"
```

Comment:
I think it must be discussed a bit. From my POV, I think it's better if the process stops brutally a gives a chance to investigate rather than having a warning lost in tons of logs that would lead to the use of not neutralized db.

Reply:
Same comment as above:
How does it integrate with SH or Saas database?
Does it prevent to creation of the duplicate?

Or do we get a duplicate with no neutralization at all?
I think I prefer to have most of the neutralization done in all case even if a single one fails.
What about a warning on the App Switcher in case of failure of any neutralization with details about it? (if possible)

Reply:
@jev-odoo thanks, I'll remove those assert then.